# Tinder Project - Speed Dating 
### What makes people say yes to a second date?

In [1]:
import pandas as pd
import numpy as np
import plotly.graph_objects as go
import plotly.express as px
import chardet 
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [2]:
file = 'Speed+Dating+Data.csv'

#detect file encoding

with open(file, 'rb') as f:
    data = f.read()

# Detect the encoding
result = chardet.detect(data)
print(result['encoding'])

# Build the Dataframe
data = pd.read_csv(file, encoding=result['encoding'])
df = pd.DataFrame(data)

df.sample(30)

Windows-1252


,iid,id,gender,idg,condtn,wave,round,position,positin1,order,...,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
8154,542,12.0,1,24,2,21,22,13,6.0,8,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2044,151,10.0,0,19,2,7,16,6,6.0,9,...,12.0,9.0,9.0,12.0,8.0,NaN,NaN,NaN,NaN,NaN
1330,93,18.0,0,35,2,4,18,18,NaN,2,...,8.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN
4051,274,2.0,1,4,2,11,21,16,16.0,10,...,9.0,9.0,9.0,9.0,9.0,8.0,8.0,8.0,9.0,8.0
6341,411,15.0,1,30,2,15,18,17,1.0,17,...,7.0,8.0,8.0,9.0,6.0,7.0,7.0,8.0,9.0,7.0
7073,479,15.0,0,28,2,19,15,7,7.0,1,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2728,200,7.0,0,13,2,9,20,18,18.0,14,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1571,107,14.0,1,28,2,4,18,1,NaN,8,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1331,93,18.0,0,35,2,4,18,18,NaN,1,...,8.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN
5293,355,15.0,0,29,2,14,18,20,20.0,11,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [3]:
df.describe(include='all')

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,...,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
count,8378.000000,8377.000000,8378.000000,8378.000000,8378.000000,8378.000000,8378.000000,8378.000000,6532.000000,8378.000000,...,3974.000000,3974.000000,3974.000000,3974.000000,3974.000000,2016.000000,2016.000000,2016.000000,2016.000000,2016.000000
unique,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,283.675937,8.960248,0.500597,17.327166,1.828837,11.350919,16.872046,9.042731,9.295775,8.927668,...,7.240312,8.093357,8.388777,7.658782,7.391545,6.810020,7.615079,7.932540,7.155258,7.048611
std,158.583367,5.491329,0.500029,10.940735,0.376673,5.995903,4.358458,5.514939,5.650199,5.477009,...,1.576596,1.610309,1.459094,1.744670,1.961417,1.507341,1.504551,1.340868,1.672787,1.717988
min,1.000000,1.000000,0.000000,1.000000,1.000000,1.000000,5.000000,1.000000,1.000000,1.000000,...,2.000000,2.000000,3.000000,2.000000,1.000000,2.000000,2.000000,4.000000,1.000000,1.000000
25%,154.000000,4.000000,0.000000,8.000000,2.000000,7.000000,14.000000,4.000000,4.000000,4.000000,...,7.000000,7.000000,8.000000,7.000000,6.000000,6.000000,7.000000,7.000000,6.000000,6.000000
50%,281.000000,8.000000,1.000000,16.000000,2.000000,11.000000,18.000000,8.000000,9.000000,8.000000,...,7.000000,8.000000,8.000000,8.000000,8.000000,7.000000,8.000000,8.000000,7.000000,7.000000
75%,407.000000,13.000000,1.000000,26.000000,2.000000,15.000000,20.000000,13.000000,14.000000,13.000000,...,8.000000,9.000000,9.000000,9.000000,9.000000,8.000000,9.000000,9.000000,8.000000,8.000000


In [4]:
df.columns

Index(['iid', 'id', 'gender', 'idg', 'condtn', 'wave', 'round', 'position',
       'positin1', 'order',
       ...
       'attr3_3', 'sinc3_3', 'intel3_3', 'fun3_3', 'amb3_3', 'attr5_3',
       'sinc5_3', 'intel5_3', 'fun5_3', 'amb5_3'],
      dtype='object', length=195)

# Identify unique member from each wave

In [5]:
unique = df[['iid', 'gender', 'age', 'race', 'goal', 'date', 'go_out']].drop_duplicates('iid')

unique['gender_label'] = unique['gender'].map({0: 'female', 1: 'male'})

race_map = {1: 'Black/African American', 2: 'European/Caucasian', 3: 'Latino/Hispanic',
            4: 'Asian/Pacific Islander', 5: 'Native American', 6: 'Other'}
unique['race_label'] = unique['race'].map(race_map)

goal_map = {1: 'Fun night out', 2: 'Meet new people', 3: 'Get a date',
            4: 'Serious relationship', 5: 'Say I did it', 6: 'Other'}
unique['goal_label'] = unique['goal'].map(goal_map)

unique

,iid,gender,age,race,goal,date,go_out,gender_label,race_label,goal_label
0,1,0,21.0,4.0,2.0,7.0,1.0,female,Asian/Pacific Islander,Meet new people
10,2,0,24.0,2.0,1.0,5.0,1.0,female,European/Caucasian,Fun night out
20,3,0,25.0,2.0,6.0,3.0,1.0,female,European/Caucasian,Other
30,4,0,23.0,2.0,1.0,5.0,1.0,female,European/Caucasian,Fun night out
40,5,0,21.0,2.0,2.0,4.0,1.0,female,European/Caucasian,Meet new people
...,...,...,...,...,...,...,...,...,...,...
8268,548,1,30.0,2.0,1.0,6.0,1.0,male,European/Caucasian,Fun night out
8290,549,1,28.0,2.0,1.0,7.0,1.0,male,European/Caucasian,Fun night out
8312,550,1,30.0,2.0,2.0,4.0,1.0,male,European/Caucasian,Meet new people
8334,551,1,27.0,1.0,1.0,6.0,3.0,male,Black/African American,Fun night out


In [6]:
print(f'Total participants: {len(unique)}')
print(f'Total speed dates recorded: {len(df)}')
print(f'Overall match rate: {df["match"].mean()*100:.1f}%')

Total participants: 551
Total speed dates recorded: 8378
Overall match rate: 16.5%


# 1. Sample Overview

### 1.1. Gender distribution

In [9]:
counts = unique['gender_label'].value_counts().reset_index()
counts.columns = ['Gender', 'Count']

fig = px.pie(
    counts, values='Count', names='Gender',
    title='Gender Distribution of Participants',
    color='Gender',
    hole=0.5
)
 
fig.update_traces(
    texttemplate='%{label}<br>%{value} (%{percent})',
    textposition='outside'
)
 
fig.update_layout(
    showlegend=False,
    title_font_size=18,
    margin=dict(t=60, b=40, l=40, r=40)
)
 
fig.show()



**Caption:** The dataset is nearly perfectly balanced with 274 women (49.7%) and 277 men (50.3%). This balance is by design — each speed dating event was organized with equal numbers of men and women. For our analysis of second date decisions, this means we can compare male and female behavior without worrying about sampling bias.

### 1.2 Age Distribution

In [10]:
f_mean, f_std = unique[unique['gender']==0]['age'].mean(), unique[unique['gender']==0]['age'].std()
m_mean, m_std = unique[unique['gender']==1]['age'].mean(), unique[unique['gender']==1]['age'].std()

In [21]:
fig = px.histogram(
    unique, x='age',facet_col='gender_label',
    nbins=20,
    color='gender_label',
    title='Age Distribution by Gender — Mean and ±1 Standard Deviation',
    labels={'age': 'Age', 'gender_label': 'Gender'}
)

for col, mean, std, color in [(1, f_mean, f_std, 'Black'),(2, m_mean, m_std, 'Black')]:
    fig.add_vline(x=mean, line_dash='solid', line_color=color, line_width=2,
                  annotation_text=f'Mean = {mean:.1f}',
                  annotation_position='top right',
                  annotation_font_color=color, col=col)
    fig.add_vline(x=mean - std, line_dash='dash', line_color=color, line_width=1.5,
                  annotation_text=f'-1 SD ({mean-std:.1f})',
                  annotation_position='bottom left',
                  annotation_font_color=color, col=col)
    fig.add_vline(x=mean + std, line_dash='dash', line_color=color, line_width=1.5,
                  annotation_text=f'+1 SD ({mean+std:.1f})',
                  annotation_position='bottom right',
                  annotation_font_color=color, col=col)    

fig.update_layout(showlegend=False)

fig.show()


**Caption:** Most participants are between 22 and 30 years old, reflecting the graduate student population at Columbia University. 

### 1.3 Race Distribution

In [33]:
counts_race = unique['race_label'].value_counts().reset_index()
counts_race.columns = ['Race', 'Count']

fig = px.bar(counts_race,
             x='Count', y='Race',
             title='Racial Composition of Participants',
             color='Count',
             color_continuous_scale='Blues',
             text='Count',             
)

fig.update_traces(textposition='outside')
fig.update_layout(
    coloraxis_showscale=False,
    yaxis=dict(categoryorder='total ascending'),
    xaxis_title='Number of participants',
    yaxis_title='',
    title_font_size=16
)

fig.show()

European/Caucasian participants are the largest group (304 out of 545 with valid data), followed by Asian/Pacific Islander (136). The sample reflects the demographics of Columbia University's graduate programs in the early 2000s. 

### 1.4 Participant Goals

In [38]:
counts_goal = unique['goal_label'].value_counts().reset_index()
counts_goal.columns = ['Goal', 'Count']
counts_goal['Percentage'] = (counts_goal['Count'] / counts_goal['Count'].sum() * 100).round(1)

fig = px.bar(
    counts_goal, x='Goal', y='Count',
    title='Why did participants sign up?',
    color='Count',
    color_continuous_scale='Pinkyl',
    text=counts_goal['Percentage'].apply(lambda x: f'{x}%')
)
fig.update_traces(textposition='inside')
fig.update_layout(
    coloraxis_showscale=False,
    xaxis_title='', yaxis_title='Number of participants',
    title_font_size=16
)
fig.show()

## 2. Match Rates

### 2.1 Overall Decision and Match Rate

### 2.2 Match Rate by Race Pairing

## 3. What Actually Drives the "Yes"?

### 3.1 Average Attribute Ratings by Match Outcome

### 3.2 Correlation of Ratings with Personal Decision

### 3.3 Stated Preferences vs Actual Predictors (by Gender)

## 4. Context and Situational Effects

### 4.1 Decision Fatigue: Match Rate by Meeting Order

### 4.2 Self-Rating vs Received Rating (Ego Calibration)

## 5. Liking, Probability Estimates, and Match

### 5.1 Overall Liking Score Distribution by Match Outcome

### 5.2 Perceived Probability of Partner's Interest

## 6. Summary